camel/toolkits/math_toolkit.py

```diff
@@ -12,22 +12,81 @@
 # limitations under the License.
 # ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
 
-from typing import List
+import subprocess
+import sys
+from typing import List, Literal, Optional, Union
 
-from camel.toolkits.base import BaseToolkit
+from camel.toolkits.code_execution import CodeExecutionToolkit
 from camel.toolkits.function_tool import FunctionTool
 from camel.utils import MCPServer
 
 
 @MCPServer()
-class MathToolkit(BaseToolkit):
-    r"""A class representing a toolkit for mathematical operations.
+class MathToolkit(CodeExecutionToolkit):
+    r"""A toolkit for performing basic and advanced mathematical computations.
 
-    This class provides methods for basic mathematical operations such as
-    addition, subtraction, multiplication, division, and rounding.
+    Attributes:
+        math_packages (list): List of math packages to use.
+
+    Args:
+        math_packages (List[str], optional): List of Python packages to use
+            for mathematical computations (e.g., `numpy`, `sympy`).
```

Comment:
Let's give the agent more hints:
```suggestion
        math_packages (List[str], optional): List of Python packages to use
            for mathematical computations (e.g., `numpy`, `sympy`, `cvxpy`,  `scipy`, `networkx`).
```